### PR TITLE
Correctly handle missing status, body, schema

### PR DESCRIPTION
**Pull request by honzajavorek**
There are multiple issues on Dredd, mainly in the realm of Swagger, which are dealing with situations when user doesn't specify expected status code, body, or schema in the API description document. This PR is trying to address such issues:

- Single `default` response in Swagger should be assumed as HTTP 200. If there are any other responses alongside the `default` one, it should be ignored completely (unsupported scenario, parser gives a warning).
- When user doesn't specify expected body or schema, make sure those are not coerced to any values. (Note there is no difference in API Elements between body provided as `undefined` or empty string.) They should be passed on to Dredd and Dredd should decide what to do with them - see https://github.com/apiaryio/dredd/issues/556#issuecomment-342879248
- Make sure media types with parameters, such as `application/json; charset=utf-8`, are correctly recognized as JSON media types.

--------

- Addressing https://github.com/apiaryio/dredd/issues/553 and https://github.com/apiaryio/dredd/issues/893
- Related to https://github.com/apiaryio/dredd/issues/883, https://github.com/apiaryio/dredd/issues/897, https://github.com/apiaryio/dredd/issues/556

------

This PR unfortunately brings some conflicts with the `honzajavorek/breaking-changes` branch, but I'll deal with it later. Changes in this PR are more important at this moment.

### CODE REVIEW

File: src/compile.coffee [hunk 1]
```diff
@@ -146,7 +148,7 @@ compileOriginExampleName = (mediaType, httpResponseElement, exampleNo) ->
     if exampleNo
       exampleName = "Example #{exampleNo}"
   else
-    statusCode = httpResponseElement.statusCode.toValue()
+    statusCode = httpResponseElement.statusCode?.toValue() or '200'
```
Comment (kylef) [inline src/compile.coffee]: Maybe `200` shouldn't be the default in some cases. If there is no schema or body then `204` may make more sense?

Comment (honzajavorek) [reply src/compile.coffee]: Empty bodies could be 205 as well... I'm not sure I want to dig into more complicated logic at this moment. No schema/body can just mean there are no assertions. Let's see how users work with this and then let's reiterate on what we can do better.

File: test/integration/compile-api-blueprint-test.coffee [hunk 2]
```diff
@@ -283,4 +282,26 @@ describe('compile() · API Blueprint', ->
       assert.equal(compilationResult.transactions[0].request.uri, '/honey?beekeeper=Pavan')
     )
   )
+
+  describe('with response without explicit status code', ->
+    compilationResult = undefined
+
+    beforeEach((done) ->
```
Comment (michalholasek) [inline test/integration/compile-api-blueprint-test.coffee]: Do we need to compile given description document before each test? Parser would not produce different results for each call, would it?

File: test/integration/compile-test.coffee [hunk 3]
```diff
@@ -675,4 +675,70 @@ describe('compile() · all API description formats', ->
...
+      )
+
+      it('produces no annotations and 2 transactions', ->
+        assert.jsonSchema(compilationResult, createCompilationResultSchema(
+          errors: 0
+          warnings: 0
+          transactions: 2
+        ))
+      )
+      it('produces transactions #1 with no body', ->
```
Comment (michalholasek) [inline test/integration/compile-test.coffee]: NIT: `produces transaction #1...` (extra 's').

File: test/integration/compile-api-blueprint-test.coffee (hunk 2 above)
Comment (honzajavorek) [reply test/integration/compile-api-blueprint-test.coffee]: I'll address this in a subsequent PR - I'll change it everywhere in these tests.

**Comment by honzajavorek** [comment]
Updated.

**Pull request merged**